common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -524,13 +540,73 @@ protected SecretKey loadSecretKey(@NonNull final KeyType keyType) throws IOExcep
                 return getSecretKey(AuthenticationSettings.INSTANCE.getSecretKeyData());
 
             case KEYSTORE_ENCRYPTED_KEY:
-                return getKey();
+                return loadKey();
         }
 
         Logger.verbose(TAG + methodName, "Unknown KeyType. This code should never be reached.");
         throw new GeneralSecurityException(ErrorStrings.UNKNOWN_ERROR);
     }
 
+    /**
+     * If needed, get the key from inactive broker (if there is one).
+     * If it fails to get a new key. It will create a new one.
+     *
+     * This function can only be invoked by Broker apps.
+     */
+    public void migrateEncryptionKeyIfNeeded() throws GeneralSecurityException, IOException {
+        final String methodName = ":migrateEncryptionKeyIfNeeded";
+
+        if (!(COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(getPackageName()) ||
```

Comment:
These are only called from broker auth paths right? As at least Company Portal currently does local auth, it wants to do broker auth ...

Reply:
Migration should only be called by Broker. yes.